src/Service/ReadmeExtractor.php

```diff
@@ -51,10 +52,13 @@ public function extractReadme(Package $package, Dist $dist): void
 
     private function loadREADME(Dist $dist): ?string
     {
-        $zipFilename = $this->distStorage->filename($dist);
+        $tmpLocalFilename = $this->createTemporaryZipFile($dist);
```

Comment:
This here I would like some comment on it. If anyone has a better idea on how to resolve this problem of reading a zip file from stream, would be nice. I saw this lib here: https://github.com/maennchen/ZipStream-PHP, but dunno if it is worth the effort to integrate with it. Also, there is always the TechDebt that comes with the lib itself.

Reply:
I don't think it will make much difference, and we still have to go through the entire archive, unless it's actually smart enough that it can only get the index of the files and then the readme itself. Either way, it can stay as it is at the moment. This will only happen once for a new package.

Reply:
Okay, I am resolving this one. This library is much more focused on streaming zip responses, so I don't think it will provide this kind of control over reading one ZIP file.